plugin/documents.py

```diff
@@ -158,6 +158,7 @@ def on_change() -> None:
         self._registration = SettingsRegistration(view.settings(), on_change=on_change)
         self._completions_task = None  # type: Optional[QueryCompletionsTask]
         self._stored_selection = []  # type: List[sublime.Region]
+        self.paste_was_run = False
```

Comment:
Private properties should start with an underscore.
Also maybe call it `_did_paste`